### PR TITLE
[env](compile)open compile check in data type serde files

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_array_serde.cpp [hunk 1]
```diff
@@ -220,19 +221,19 @@ Status DataTypeArraySerDe::serialize_one_cell_to_hive_text(
 
 void DataTypeArraySerDe::write_one_cell_to_jsonb(const IColumn& column, JsonbWriter& result,
                                                  Arena* mem_pool, int32_t col_id,
-                                                 int row_num) const {
-    result.writeKey(col_id);
+                                                 int64_t row_num) const {
+    result.writeKey(cast_set<JsonbKeyValue::keyid_type>(col_id));
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_array_serde.cpp:225]: add a comment  for the cast_set, seems `JsonbKeyValue::keyid_type` is uint16_t

File: be/src/vec/data_types/serde/data_type_bitmap_serde.cpp [hunk 2]
```diff
@@ -52,17 +53,17 @@ Status DataTypeBitMapSerDe::deserialize_one_cell_from_json(IColumn& column, Slic
...
 
-Status DataTypeBitMapSerDe::write_column_to_pb(const IColumn& column, PValues& result, int start,
-                                               int end) const {
+Status DataTypeBitMapSerDe::write_column_to_pb(const IColumn& column, PValues& result,
+                                               int64_t start, int64_t end) const {
     auto ptype = result.mutable_type();
     ptype->set_id(PGenericType::BITMAP);
     auto& data_column = assert_cast<const ColumnBitmap&>(column);
-    int row_count = end - start;
+    auto row_count = cast_set<int>(end - start);
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_bitmap_serde.cpp:61]: why here need cast ?  use int64_t?

File: be/src/vec/data_types/serde/data_type_bitmap_serde.cpp [hunk 3]
```diff
@@ -81,12 +82,12 @@ Status DataTypeBitMapSerDe::read_column_from_pb(IColumn& column, const PValues&
...
                                                   Arena* mem_pool, int32_t col_id,
-                                                  int row_num) const {
+                                                  int64_t row_num) const {
     auto& data_column = assert_cast<const ColumnBitmap&>(column);
-    result.writeKey(col_id);
+    result.writeKey(cast_set<JsonbKeyValue::keyid_type>(col_id));
     auto bitmap_value = const_cast<BitmapValue&>(data_column.get_element(row_num));
     // serialize the content of string
-    auto size = bitmap_value.getSizeInBytes();
+    auto size = cast_set<uint32_t>(bitmap_value.getSizeInBytes());
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_bitmap_serde.cpp:90]: why here need cast  ? if the byte over uint32_t is a danger behaviror

File: be/src/vec/data_types/serde/data_type_datetimev2_serde.cpp [hunk 4]
```diff
@@ -162,7 +166,7 @@ void DataTypeDateTimeV2SerDe::read_column_from_arrow(IColumn& column,
             // convert second
             v.from_unixtime(utc_epoch / divisor, ctz);
             // get rest time
-            v.set_microsecond(utc_epoch % divisor);
+            v.set_microsecond(cast_set<uint32_t>(utc_epoch % divisor));
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_datetimev2_serde.cpp:169]: why not condier change the api ?

File: be/src/vec/data_types/serde/data_type_decimal_serde.h [hunk 5]
```diff
@@ -134,8 +136,8 @@ class DataTypeDecimalSerDe : public DataTypeSerDe {
 
 template <typename T>
 Status DataTypeDecimalSerDe<T>::write_column_to_pb(const IColumn& column, PValues& result,
-                                                   int start, int end) const {
-    int row_count = end - start;
+                                                   int64_t start, int64_t end) const {
+    auto row_count = cast_set<int>(end - start);
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_decimal_serde.h:140]: int64?

File: be/src/vec/data_types/serde/data_type_bitmap_serde.cpp (hunk 2 above)
Comment (zhangstar333) [reply be/src/vec/data_types/serde/data_type_bitmap_serde.cpp:61]: result.mutable_bytes_value()->Reserve(row_count);
`Reserve` need int argument, 
and the function is call as block batch, so int of row_count is ok.

File: be/src/vec/data_types/serde/data_type_datetimev2_serde.cpp (hunk 4 above)
Comment (zhangstar333) [reply be/src/vec/data_types/serde/data_type_datetimev2_serde.cpp]: done

File: be/src/vec/data_types/serde/data_type_bitmap_serde.cpp (hunk 3 above)
Comment (zhangstar333) [reply be/src/vec/data_types/serde/data_type_bitmap_serde.cpp]: done

File: be/src/vec/data_types/serde/data_type_array_serde.cpp (hunk 1 above)
Comment (zhangstar333) [reply be/src/vec/data_types/serde/data_type_array_serde.cpp:226]: done

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_datev2_serde.cpp [hunk 6]
```diff
@@ -81,18 +82,18 @@ Status DataTypeDateV2SerDe::deserialize_one_cell_from_json(IColumn& column, Slic
...
-        int32_t daynr = binary_cast<UInt32, DateV2Value<DateV2ValueType>>(col_data[i]).daynr() -
-                        date_threshold;
+        auto daynr = binary_cast<UInt32, DateV2Value<DateV2ValueType>>(col_data[i]).daynr() -
+                     date_threshold;
         if (null_map && (*null_map)[i]) {
             checkArrowStatus(date32_builder.AppendNull(), column.get_name(),
                              array_builder->type()->name());
         } else {
-            checkArrowStatus(date32_builder.Append(daynr), column.get_name(),
+            checkArrowStatus(date32_builder.Append(cast_set<int>(daynr)), column.get_name(),
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_datev2_serde.cpp:96]: here is a for loop need check every value ?

- [ ]

File: be/src/vec/data_types/serde/data_type_number_serde.h [hunk 7]
```diff
@@ -121,14 +122,14 @@ Status DataTypeNumberSerDe<T>::read_column_from_pb(IColumn& column, const PValue
         column.resize(old_column_size + arg.uint32_value_size());
         auto& data = assert_cast<ColumnType&>(column).get_data();
         for (int i = 0; i < arg.uint32_value_size(); ++i) {
-            data[old_column_size + i] = arg.uint32_value(i);
+            data[old_column_size + i] = cast_set<T, uint32_t>(arg.uint32_value(i));
```
Comment (HappenLee) [inline be/src/vec/data_types/serde/data_type_number_serde.h:125]: for loop should not do the check? and should not check in T is UINT32

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Mryange — APPROVED** [review]

**Pull request merged**